src/parser/SparqlParser.cpp

```diff
@@ -338,81 +338,74 @@ void SparqlParser::parseWhere(ParsedQuery* query,
       currentPattern->_children.emplace_back(std::move(values));
       lexer_.accept(".");
     } else {
-      std::string subject;
-      if (lastSubject.empty()) {
-        if (lexer_.accept(SparqlToken::Type::VARIABLE)) {
-          subject = lexer_.current().raw;
-          query->registerVariableVisibleInQueryBody(lexer_.current().raw);
-        } else {
-          lexer_.expect(SparqlToken::Type::IRI);
-          subject = lexer_.current().raw;
-        }
-      } else {
-        subject = lastSubject;
-        lastSubject.clear();
-      }
-
-      std::string predicate;
-      if (lastPredicate.empty()) {
-        if (lexer_.accept(SparqlToken::Type::VARIABLE)) {
-          predicate = lexer_.current().raw;
-          query->registerVariableVisibleInQueryBody(lexer_.current().raw);
-        } else if (lexer_.accept(SparqlToken::Type::A_RDF_TYPE_ALIAS)) {
-          predicate = "<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>";
+      auto iri = [](const Iri& iri) { return TripleComponent{iri.toSparql()}; };
+      auto blankNode = [](const BlankNode& blankNode) -> TripleComponent {
+        return blankNode.toSparql();
+      };
+      auto literal = [](const Literal& literal) {
+        // Problem: ql:contains-word causes the " to be stripped.
+        // TODO: Move stripAndLowercaseKeywordLiteral out to this point or
+        //  rewrite the Turtle Parser s.t. this code can be integrated into the
+        //  visitor. In this case the turtle parser should output the
+        //  corresponding modell class.
+        if (literal.toSparql()[0] == '"') {
+          return TurtleStringParser<TokenizerCtre>::parseTripleObject(
+              literal.toSparql());
         } else {
-          // Assume the token is a predicate path. This will be verified
-          // separately later.
-          lexer_.expandNextUntilWhitespace();
-          lexer_.accept();
-          predicate = lexer_.current().raw;
+          return TripleComponent{literal.toSparql()};
```

Comment:
This part imo is a huge red flag. Previously the TurtleParsing was done first. Then the check for the contains CONTAINS_WORD_PREDICATE which removes the quotation marks. Now it is the other way round. Because the check for the contains word can be integrated into the Visitor. The TurtleParsing cannot (at least easily) because it emits a very general type (TripleComponent).
If the contains word removed the quotation marks the TurtleParsing fails. Thats why check is there. I outlined two possible solutions I can think of in the commont above.